src/core/lib/gprpp/thd_posix.cc

```diff
@@ -48,6 +49,24 @@ struct thd_arg {
   bool tracked;
 };
 
+const size_t page_size = static_cast<size_t>(sysconf(_SC_PAGESIZE));
```

Comment:
Since this is only used by RoundUpToPageSize, I would recommend declaring it as a static variable there so that its scope is clear and so as to avoid any issues with global variables needing to be initialized with constexpr function (which sysconf is not, afaik).

Reply:
Using a static variable here would cause a linking error as I tried before:
```
/var/local/git/grpc/libs/dbg/libgrpc.a(thd_posix.o): In function `RoundUpToPageSize':
/var/local/git/grpc/src/core/lib/gprpp/thd_posix.cc:53: undefined reference to `__cxa_guard_acquire'
/var/local/git/grpc/src/core/lib/gprpp/thd_posix.cc:53: undefined reference to `__cxa_guard_release'
```
(https://source.cloud.google.com/results/invocations/07ba970e-6c1e-4ebe-8dd9-180218eee5c5/targets/github%2Fgrpc%2Faggregate_tests/tests).
The reason for this can be explained by https://www.geeksforgeeks.org/g-fact-80/

Reply:
That article is about C, though. This code is C++. static doesn't have to be initialized that way in C++. See for example the compiler results window for https://godbolt.org/z/3qyTTd

Reply:
Ah, seeing the error link, though, the issue is a different one. This is in core but function-level static initialization uses standard-library locks for making sure that static variables are only initialized once, and we can't use standard-library in core.

So I'll approve this PR but I'm going to ask you to add a TODO comment on that line indicating that we should find a way to move this to a function-level static or remove the use of a non-constexpr initializer when possible.